File: 백준/Silver/1094. 막대기/막대기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int input = Integer.parseInt(br.readLine());
        int count = 0;
        for (char c : Integer.toBinaryString(input).toCharArray()) {
            if (c == '1') count++;
        }
        System.out.println(count);
    }
}

File: 백준/Silver/1094. 막대기/막대기.kt
import java.io.BufferedReader

fun main() = with(BufferedReader(System.`in`.bufferedReader())) {
    val n = readLine().toInt()

    // n을 이진법으로 만든다
    val binary = n.toString(2)

    val count = binary.split("")
        .filter { it == "1" }
        .count()
    println("${count}")
}